File: back/src/main/java/csh/back/domain/vote/vote/dto/response/VoteFindResponse.java
package csh.back.domain.vote.vote.dto.response;

public record VoteFindResponse(
        Long tripPlaceId,
        String place,
        Long count,
        boolean isVoted) {
    public static VoteFindResponse of(Long tripPlaceId, String place, Long count, boolean isVoted) {
        return new VoteFindResponse(tripPlaceId, place, count, isVoted);
    }
}

File: back/src/main/kotlin/csh/back/domain/vote/vote/dto/response/VoteFindResponse.kt
package csh.back.domain.vote.vote.dto.response

data class VoteFindResponse(
    val tripPlaceId: Long?,
    val place: String,
    val count: Long,
    val isVoted: Boolean,
    val voters: List<VoterResponse>,
) {
    companion object {
        @JvmStatic
        fun of(tripPlaceId: Long?, place: String, count: Long, isVoted: Boolean, voters: List<VoterResponse>): VoteFindResponse =
            VoteFindResponse(tripPlaceId, place, count, isVoted, voters)
    }
}
